core/impl/src/main/java/com/blazebit/persistence/impl/JoinManager.java

```diff
@@ -709,8 +704,10 @@ private static boolean isNoJoinValuesNode(JoinNode rootNode) {
                 } else {
                     if (type instanceof EntityType<?>) {
                         sb.append(((EntityType) type).getName());
-                        if (rootNode.getValuesIdName() != null) {
-                            sb.append('.').append(rootNode.getValuesIdName());
+                        Set<String> valuesIdNames = rootNode.getValuesIdNames();
+                        if (valuesIdNames != null && ! valuesIdNames.isEmpty()) {
+                            String next = valuesIdNames.iterator().next();
+                            sb.append(".").append(next);
```

Comment:
I still want to look into this method to see how it can be adapted for compound keys. After that, this PR is ready to be merged.